sympy/functions/combinatorial/tests/test_comb_numbers.py

```diff
@@ -398,6 +398,26 @@ def test_genocchi():
     assert genocchi(4 * n - 2).is_negative
 
 
+def test_partition():
+    partition_nums = [1, 1, 2, 3, 5, 7, 11, 15, 22]
+    for n, p in enumerate(partition_nums):
+        assert partition(n) == p
+
+    x = Symbol('x')
+    y = Symbol('y', real=True)
+    m = Symbol('m', integer=True)
+    n = Symbol('n', integer=True, negative=True)
+    p = Symbol('p', integer=True, nonnegative=True)
+    assert partition(m).is_integer
+    assert not partition(m).is_negative
+    assert partition(m).is_nonnegative
+    assert partition(n).is_zero
+    assert partition(p).is_positive
+    raises(ValueError, lambda: partition(x))
```

Comment:
Hmm...we should probable not raise an error unless arg.is_integer is False. This would be consistent with the return of None from `is_integer` if the arg is not an integer.

Reply:
Okay, so should I remove the lines 416 through 418?

Reply:
It might also be possible to just return `arg.is_integer` to ensure consistency.